python/lsst/source/injection/consolidate_injected_catalogs.py

```diff
@@ -0,0 +1,283 @@
+
+from __future__ import annotations
+
+__all__ = [
+    "ConsolidateInjectedCatalogsConnections",
+    "ConsolidateInjectedCatalogsConfig",
+    "ConsolidateInjectedCatalogsTask",
+]
+from typing import cast
+
+import numpy as np
+from astropy.table import Table, vstack
+from astropy.table.column import MaskedColumn
+from lsst.geom import SpherePoint, degrees
+from lsst.pex.config import Field
+from lsst.pipe.base import PipelineTask, PipelineTaskConfig, PipelineTaskConnections, Struct
+from lsst.pipe.base.connections import InputQuantizedConnection
+from lsst.pipe.base.connectionTypes import Input, Output, PrerequisiteInput
+from lsst.skymap import BaseSkyMap
+from lsst.skymap.ringsSkyMap import RingsSkyMap
+from smatch.matcher import Matcher  # type: ignore [import-not-found]
+
+
+class ConsolidateInjectedCatalogsConnections(
+    PipelineTaskConnections,
+    dimensions=("instrument", "skymap", "tract"),
+    defaultTemplates={
+        "injected_prefix": "injected_",
+    },
+):
+    """Base connections for source injection tasks."""
+
+    input_catalogs = PrerequisiteInput(
+        doc="Per-patch and per-band injected catalogs to draw inputs from.",
+        name="{injected_prefix}deepCoadd_catalog",
+        dimensions=("skymap", "tract", "patch", "band"),
+        storageClass="ArrowAstropy",
+        minimum=0,
+        multiple=True,
+    )
+    skyMap = Input(
+        doc="Input definition of geometry/bbox and projection/wcs for warped exposures",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+    output_catalog = Output(
+        doc="Per-tract multiband catalog of injected sources.",
+        name="{injected_prefix}deepCoadd_catalog_tract",
+        storageClass="ArrowAstropy",
+        dimensions=("skymap", "tract"),
+    )
+
+
+class ConsolidateInjectedCatalogsConfig(  # type: ignore [call-arg]
+    PipelineTaskConfig, pipelineConnections=ConsolidateInjectedCatalogsConnections
+):
+    """Base configuration for source injection tasks."""
+
+    col_ra = Field[str](
+        doc="Column name for right ascension (in degrees).",
+        default="ra",
+    )
+    col_dec = Field[str](
+        doc="Column name for declination (in degrees).",
+        default="dec",
+    )
+    col_mag = Field[str](
+        doc="Column name for magnitude.",
+        default="mag",
+    )
+    col_source_type = Field[str](
+        doc="Column name for the source type used in the input catalog. Must match one of the surface "
+        "brightness profiles defined by GalSim.",
+        default="source_type",
+    )
+
+
+class ConsolidateInjectedCatalogsTask(PipelineTask):
+    """Class for combining all tables in a collection of input catalogs
+    into one table.
+    """
+
+    _DefaultName = "consolidateInjectedCatalogsTask"
+    ConfigClass = ConsolidateInjectedCatalogsConfig
+
+    def runQuantum(self, butler_quantum_context, input_refs, output_refs):
```

Comment:
I think butler_quantum_context is more commonly butlerQC but I don't know if we have a standard.

Reply:
The base class implementation has:

```
    def runQuantum(
        self,
        butlerQC: QuantumContext,
        inputRefs: InputQuantizedConnection,
        outputRefs: OutputQuantizedConnection,
    ) -> None:
        """Do butler IO and transform to provide in memory
        objects for tasks `~Task.run` method.

        Parameters
        ----------
        butlerQC : `QuantumContext`
            A butler which is specialized to operate in the context of a
            `lsst.daf.butler.Quantum`.
```

Reply:
changed this to butlerQC, thanks